bot/exts/evergreen/fun.py

```diff
@@ -87,6 +92,93 @@ def conversion_func(text: str) -> str:
             converted_text = f">>> {converted_text.lstrip('> ')}"
         await ctx.send(content=converted_text, embed=embed)
 
+    @commands.group(name="caesarcipher", aliases=("caesar", "cc",))
+    async def caesarcipher_group(self, ctx: Context) -> None:
+        """
+        Translates a message using the Caesar Cipher.
+
+        See `decrpyt`, `encrypt`, and `info` subcommands.
+        """
+        if ctx.invoked_subcommand is None:
+            await ctx.invoke(self.bot.get_command("help"), "caesarcipher")
+
+    @caesarcipher_group.command(name="info")
+    async def caesarcipher_info(self, ctx: Context) -> None:
+        """Information about the Caesar Cipher."""
+        embed = Embed.from_dict(self._caesar_cipher_embed)
+        embed.colour = Colours.dark_green
+
+        await ctx.send(embed=embed)
+
+    @staticmethod
+    async def _caesar_cipher(ctx: Context, offset: int, msg: Union[Message, str]) -> None:
+        """
+        Given an integer `offset`, translates and sends the given `msg`.
+
+        A positive `offset` will cause the letters to shift right, while
+        a negative `offset` will cause the letters to shift left.
+
+        Also accepts a valid Discord Message ID or link.
+        """
+        def caesar_func(text: str) -> Iterable[str]:
+            """Implements a lazy Caesar Cipher algorithm."""
+            for char in text:
+                if not char.isascii() or not char.isalpha() or char.isspace():
+                    yield char
+                    continue
+
+                case_start = 65 if char.isupper() else 97
+                true_offset = (ord(char) - case_start + offset) % 26
+
+                yield chr(case_start + true_offset)
+
+        def conversion_func(text: str) -> str:
+            """Encrypts the given string using the Caesar Cipher."""
+            return "".join(caesar_func(text))
+
+        is_message = isinstance(msg, Message)
+
+        text = msg.content if is_message else msg
+        embed = msg.embeds[0] if is_message and msg.embeds else None
+
+        if embed is not None:
+            embed = Fun._convert_embed(conversion_func, embed)
+
+        converted_text = conversion_func(text)
+
+        if converted_text:
+            converted_text = f">>> {converted_text.lstrip('> ')}"
+
+        await ctx.send(content=converted_text, embed=embed)
+
+    @caesarcipher_group.command(name="encrypt", aliases=("rightshift", "rshift", "enc",))
+    async def caesarcipher_encrypt(self, ctx: Context, offset: int, *, msg: Union[Message, str]) -> None:
+        """
+        Given a positive integer `offset`, encrypt the given `msg`.
+
+        Performs a right shift of the letters in the message.
+
+        Also accepts a valid Discord Message ID or link.
+        """
+        if offset < 0:
+            await ctx.send(":no_entry: Cannot use a negative offset.")
```

Comment:
Because this repeat on L177, I think maybe is better to move it to `_caesar_cipher`.

Reply:
`caesarcipher_decrypt` passes a negated `offset` to the `_caesar_cipher` method to perform a left shift though, because of how I've written the formula for the `true_offset` in L131.

Reply:
I've decided to refactor the translation methods as this was a rather trivial change. Thanks for the suggestion.